### ISSUE TITLE
Storage API returns 504 error

### ISSUE THREAD

**Issue by Sjord** [Sjord/growatt_api_client#16]
The [storage API](https://github.com/Sjord/growatt_api_client/pull/13) doesn't seem to work well for everyone. There are two reports of it returning 504:

* @wynandmyriota [here](https://github.com/Sjord/growatt_api_client/pull/13#issuecomment-692472950)
* @schimmelbirne [here](https://github.com/Sjord/growatt_api_client/issues/12#issuecomment-629094940)

**Comment by Sjord** [comment]
`get_storage_params` only takes a storage serial number. That method still works for certain storage_sns, so it's not always broken. It doesn't work for some other storage_sns, which make me suspect the storage_sn is incorrect. Could it be that the serial returned by `get_all_device_list` is incorrect, or doesn't correspond to the storage unit?

Could you take a look at the screenshots in [this comment](https://github.com/Sjord/growatt_api_client/issues/12#issuecomment-618817514)? There, TEE194904F is the storage sn which seems to work correctly with the API. What does it say for you there? Is it the same serial as returned by `get_all_device_list`?

**Comment by wynandmyriota** [comment]
Hi Sjord,

Thank you very much for responding.

I can confirm that the serial number I get from `get_all_device_list` is the same as what is displayed on the web server.

![image](https://user-images.githubusercontent.com/32625648/93292522-e86a1980-f824-11ea-8c06-b3705ea6d45b.png)

**Comment by Sjord** [comment]
Apparently, the SPA3000 doesn't use the storage API, but the newSpaApi.do endpoint. Currently there are no methods in this project that call that endpoint.

**Comment by Sjord** [comment]
I have made a few methods. Please check out the storage branch, and try something like this:

```
with GrowattApi() as api:
    api.login(username, password)
    spa = api.spa(plant_id, device_sn)
    print(spa.get_energy_overview())
    print(spa.get_system_status())
```

**Comment by wynandmyriota** [comment]
I've tried the two API calls and it works perfectly!!!! Thank you very much. Can I ask how did you manage to figure this out?

I have another question regarding the data packets received from the ShinLan box, but don't think this is the place to have the discussion. Is there another way for me to contact you?

**Comment by Sjord** [comment]
> I've tried the two API calls and it works perfectly!!!!

Nice to hear it works.

> Can I ask how did you manage to figure this out?

I [intercepted the traffic from the mobile app](https://www.sjoerdlangkemper.nl/2020/05/06/testing-android-apps-on-a-virtual-machine/), and replaced your data in the response to get_all_device_list. That made the app think I have a SPA3000, and it performs the calls to the SPA endpoint.

> I can't get the get_energy_prod_and_cons_data call to work though.

Apparently, this uses a different value for type/timespan than the other methods. I updated this function, please pull the change and try again.

> I have another question regarding the data packets received from the ShinLan box

I don't think I can help you with this, but you are free to send me an [email](mailto:sjoerd-github@linuxonly.nl).

**Comment by wynandmyriota** [comment]
Hi,

Thank you very much for your time! I've tried the new `get_energy_prod_and_cons_data` call as follow:

`    date = datetime.date.today()
    print(date)
    print(spa.get_energy_prod_and_cons_data(date, Timespan.day))`

But still get the following error:
`    raise GrowattApiError("Request failed: %s" % response)
growatt.GrowattApiError: Request failed: <Response [500]>`

**Comment by wynandmyriota** [comment]
Hi,

I've been using the hybrid API calls (newMixApi) for a few months without any issues. A few days ago this stopped working and none of my API calls work.

Is anyone experiencing the same issue with any other API call? I've checked the network captured data and nothing seems to have change.

Any help will be greatly appreciate.

**Comment by Sjord** [comment]
Yes, [other people were having the same problem](https://github.com/Sjord/growatt_api_client/issues/10#issuecomment-918495860). I fixed it in the master branch, but you didn't get the fix because you are on another branch. I'll fix this soon.

**Comment by wynandmyriota** [comment]
Thank you for letting me know.  May I ask what the issue was?

**Comment by wynandmyriota** [comment]
It is the API for logging in to the server.

I changed my code to use newLoginAPI and it's  working again!

**Comment by Sjord** [comment]
That's right. I merged the changes from master into the storage branch, so this should now work correctly on both branches.

**Comment by wynandmyriota** [comment]
Thank you very much!

**Comment by Sjord** [comment]
@WynandPrivate Maybe this issue is relevant for you: [Library is operating inconsistently after recent changes on the Growatt server side · Issue #36 · indykoning/PyPi\_GrowattServer](https://github.com/indykoning/PyPi_GrowattServer/issues/36)

**Comment by WynandPrivate** [comment]
Hi @Sjord Thank you for letting me know. So far so good for me. I've not experienced any issues yet. Will keep an eye out!
